nibabel/streamlines/tests/test_array_sequence.py

```diff
@@ -0,0 +1,255 @@
+        # Need to make sure we extend only the data we need.
+        seq = SEQ_DATA['seq'].copy()  # Copy because of in-place modification.
+        seq.extend(ArraySequence(new_data)[::2])
+        check_arr_seq(seq, SEQ_DATA['data'] + new_data[::2])
+
+        # Test extending an empty ArraySequence
+        seq = ArraySequence()
+        seq.extend(ArraySequence())
+        check_empty_arr_seq(seq)
+
+        seq.extend(SEQ_DATA['seq'])
+        check_arr_seq(seq, SEQ_DATA['data'])
+
+        # Extend with elements of different shape.
+        data = generate_data(nb_arrays=10,
+                             common_shape=SEQ_DATA['seq'].common_shape*2,
+                             rng=SEQ_DATA['rng'])
+        seq = SEQ_DATA['seq'].copy()  # Copy because of in-place modification.
+        assert_raises(ValueError, seq.extend, data)
+
+    def test_arraysequence_getitem(self):
+        # Get one item
+        for i, e in enumerate(SEQ_DATA['seq']):
+            assert_array_equal(SEQ_DATA['seq'][i], e)
+
+        # Get all items using indexing (creates a view).
+        indices = list(range(len(SEQ_DATA['seq'])))
+        seq_view = SEQ_DATA['seq'][indices]
+        check_arr_seq_view(seq_view, SEQ_DATA['seq'])
+        # We took all elements so the view should match the original.
+        check_arr_seq(seq_view, SEQ_DATA['seq'])
+
+        # Get multiple items using ndarray of dtype integer.
+        for dtype in [np.int8, np.int16, np.int32, np.int64]:
+            seq_view = SEQ_DATA['seq'][np.array(indices, dtype=dtype)]
+            check_arr_seq_view(seq_view, SEQ_DATA['seq'])
+            # We took all elements so the view should match the original.
+            check_arr_seq(seq_view, SEQ_DATA['seq'])
+
+        # Get multiple items out of order (creates a view).
+        SEQ_DATA['rng'].shuffle(indices)
+        seq_view = SEQ_DATA['seq'][indices]
+        check_arr_seq_view(seq_view, SEQ_DATA['seq'])
+        check_arr_seq(seq_view, np.asarray(SEQ_DATA['data'])[indices])
+
+        # Get slice (this will create a view).
+        seq_view = SEQ_DATA['seq'][::2]
+        check_arr_seq_view(seq_view, SEQ_DATA['seq'])
+        check_arr_seq(seq_view, SEQ_DATA['data'][::2])
+
+        # Use advanced indexing with ndarray of data type bool.
+        selection = np.array([False, True, True, False, True])
+        seq_view = SEQ_DATA['seq'][selection]
+        check_arr_seq_view(seq_view, SEQ_DATA['seq'])
+        check_arr_seq(seq_view, np.asarray(SEQ_DATA['data'])[selection])
+
+        # Test invalid indexing
+        assert_raises(TypeError, SEQ_DATA['seq'].__getitem__, 'abc')
+
+    def test_arraysequence_repr(self):
+        # Test that calling repr on a ArraySequence object is not falling.
+        repr(SEQ_DATA['seq'])
+
+    def test_save_and_load_arraysequence(self):
+        # Test saving and loading an empty ArraySequence.
+        with tempfile.TemporaryFile(mode="w+b", suffix=".npz") as f:
```

Comment:
This might error on Windows - did you check already?   Could use ``InTemporaryDirectory`` instead.